### ISSUE TITLE
Documentation on the publish command

### ISSUE THREAD

**Issue by johnynek** [coursier/coursier#1489]
I didn’t see any documentation on the publish command other than the help.

I have questions such as:

In sonatype there are a number of checks that must be passed by the repo (having java docs, sources, signatures, and probably more). Is there an example that shows the full process of doing a publish to sonatype?

**Comment by alexarchambault** [comment]
The `publish` command severely lacks documentation… I'm afraid [my talk at Scala Days 2019](https://www.youtube.com/watch?v=b06iQB_It8o) is still the best documentation for now…

It's not very well-tested either (much less than the `install` command, added around the same time), which is why it's not too advertized yet…

And it now lags behind what [sbt-sonatype](https://github.com/xerial/sbt-sonatype) can do - sbt-sonatype can upload the artifacts as a single bundle, which tends to be much more reliable than uploading the artifacts one-by-one.

About the Sonatype checks, it would be nice if the `publish` command was checking them before uploading, but it doesn't for now.

If you plan to use it without sbt, for now, it only accepts one other kind of input: a directory with the same structure as a Maven repository.

**Comment by alexarchambault** [comment]
A quick how-to-use, with Maven-like directory as input, would be
```
$ coursier publish ./initial-repo -r https://artifacts.foo.com/maven --gpg --organization org.foo --version 1.0.1
```

This command takes `./initial-repo` as input, which should contain files like
```
org/other/core_2.13/1.0/core_2.13-1.0.pom
org/other/core_2.13/1.0/core_2.13-1.0.jar
org/other/core_2.13/1.0/core_2.13-1.0-sources.jar
```

The command above:
- adjust the organization and version,
- adds signatures,
- adds checksums,
- adds `maven-metadata.xml` files,
and pushes it to `https://artifacts.foo.com/maven`.

**Comment by alexarchambault** [comment]
To test, the destination can be a local directory too, like `-r ./test-repo` (must start with `/` or `./`).

**Comment by johnynek** [comment]
Thanks for the notes. Is it the case we also need to generate the pom file in the directory?

**Comment by alexarchambault** [comment]
Yes, the POMs are needed for now. If necessary, `--organization` and `--version` update the organization and version.

**Comment by johnynek** [comment]
note: https://github.com/coursier/coursier/issues/1513 also has some examples on publishing.
